library/yum_versionlock.py

```diff
@@ -15,117 +15,265 @@
+        else:
+            versionlock_pkg['different'].append(locked)
+    return versionlock_pkg
 
-def remove_package_versionlock(module, package):
-    """ Remove package from yum versionlock """
-    rc_code, out, err = module.run_command("%s -q versionlock delete %s"
-                                           % (YUM_BINARY, package))
-    if rc_code is 0:
-        changed = True
-        return changed
-    else:
-        module.fail_json(msg="Error: " + str(err) + str(out))
+def check_state_pkg(package, List_to_check, versionlock_packages, check_type):
+    """ Verify that all desired installed packages are locked """
+    state_pkg = dict()
+    state_pkg['present'] = []
+    state_pkg['missing'] = []
+    # For debug purpose you can include the listed packages in the debug
+    #state_pkg['listed'] = []
+    missing_pkg_version = None
+    for is_checked in List_to_check:
+        if is_checked != 'Installed Packages' and is_checked != 'Available Packages':
+            # Split to concatenate name & version
+            pkg_split = is_checked.split()
+            if missing_pkg_version:
+                pkg_name = missing_pkg_version
+                pkg_version = pkg_split[0]
+                missing_pkg_version = None
+            else:
+                pkg_name = pkg_split[0].split('.')[0]
+                try:
+                    pkg_version = pkg_split[1]
+                except IndexError:
+                    missing_pkg_version = pkg_split[0].split('.')[0]
+            if pkg_version:
+                pkg_vers = pkg_name + '-' + pkg_version
+                # For debug purpose you can include the listed packages in the debug
+                #state_pkg['listed'].append(pkg_vers)
+                if check_type == 'installed':
+                    regex_search = re.compile('^([0-9]+:)?%s-([0-9]+:)?%s\.\*'%(pkg_name,pkg_version))
+                else:
+                    regex_search = re.compile('^\!([0-9]+:)?%s-([0-9]+:)?%s\.\*'%(pkg_name,pkg_version))
+                    # Init flag is_present
+                is_present = False
+                # search for the installed package in versionlock list
+                for locked in versionlock_packages:
+                    if regex_search.search(locked):
+                        is_present = True
+                if is_present:
+                    state_pkg['present'].append(pkg_vers)
+                else:
+                    state_pkg['missing'].append(pkg_vers)
+    return state_pkg
 
 
 def main():
-    """ start main program to add/remove a package to yum versionlock"""
+    """ start main program to add/exclude/delete a package to versionlock"""
+    default_mgr = __file__.split('_')[1]
     module = AnsibleModule(
         argument_spec=dict(
-            state=dict(required=True, type='str'),
-            package=dict(required=True, type='str'),
+            state=dict(required=True, type='str', choices=['present', 'excluded', 'absent', 'clear']),
```

Comment:
To match DOC (line 66) "excluded" should be set to "exclude" to prevent missleading error messages:
fatal: [127.0.0.1]: FAILED! => {"changed": false, "msg": "value of state must be one of: present, excluded, absent, clear, got: exclude"}

Reply:
Actually, the code is working with excluded, I will update the doc to match it.